### ISSUE TITLE
Plotting issue using Matplotlib 3.3.0

### ISSUE THREAD

**Issue by sirichongchitnan** [brinckmann/montepython_public#154]
Hi,
When plotting, I encounter this when trying to plot the posteriors of more than 2 parameters:

ValueError: The number of FixedLocator locations (5), usually from a call to set_ticks, does not match the number of ticklabels (1).

A bit of googling suggests that this is an issue with Matplotlib 3.3.0 (but not 3.2.x). I don't know if anyone else has encountered this issue?

Thanks,
Siri

**Comment by brinckmann** [comment]
Hi,

Are you passing a .plot file with --extra or using default MontePython plotting settings?

Best,
Thejs

**Comment by sirichongchitnan** [comment]
Both gave the same result Thejs. Even running info without --extra gives the same error.

**Comment by brinckmann** [comment]
Hi Siri,

It seems something weird is happening with the automatically generated tick labels. The following is an option I'll be including in the public version soon. I just need to test it for Python 3 (if you're using Python 3 you may have to check if for compatibility):
```
        # Ticks might have been set by the user (overrides ticks set by force_limits
        if name in conf.set_ticks.iterkeys():
            for info in information_instances:
                if not info.ignore_param:
                    while len(info.set_ticks[name]) < info.ticknumber:
                        info.set_ticks[name].append(info.set_ticks[name][-1])
                    if len(info.set_ticks[name]) > info.ticknumber:
                        warnings.warn('Number of ticks from info.set_ticks exceeds '
                                      'info.ticknumber. Skipping the last ticks.')
                        info.set_ticks[name] = info.set_ticks[name][:info.ticknumber]
                    info.ticks[info.native_index] = info.set_ticks[name]
```
You should add it on line 542 of `montepython/analyze.py`

Then you will be able to define a dictionary in the plot file, specifying an entry for whichever parameters you want to manually set the ticks for. So you should figure out which parameters are problematic and define the ticks manually, e.g.:
`info.set_ticks = {'omega_cdm': [0.1, 0.11, 0.12, 0.13, 0.14], 'n_s': [0.95, 9.96, 0.97, 0.98, 0.99]}`
You can give it a shorter list than specified with `info.ticknumber` (it will autofill with more copies of the last entry). If you provide more entries than `info.ticknumber`, it will skip the last ones.

Let me know if it solves the problem. I suspect the problem may be related to a parameter that isn't converged, which would also be nice to check so we know if it's a more general problem.

Best,
Thejs

**Comment by sirichongchitnan** [comment]
Thanks Thejs.
It looks like your fix has eliminated the set_ticks error. However, as I am using Python 3, I found that I needed to change the first line of code in your fix above to:
 if name in conf.set_ticks.keys():
(rather than iterkeys)

**Comment by brinckmann** [comment]
Awesome! Thanks for reporting back the Py3 compatibility change, that's helpful!

Best,
Thejs

**Comment by Gustavobflh** [comment]
Hello everyone,

I was having the same issue and i fixed it just change the argument in line 990 and 999  from [''] to []. So, i think that is the incompatibility. More precisely:
ax2dsub.set_xticklabels(['']) -> ax2dsub.set_xticklabels([])
ax2dsub.set_yticklabels(['']) -> ax2dsub.set_yticklabels([])

Best,
          Gustavo

**Comment by lesgourg** [comment]
One of my students had the same problem. The simple fix described by @Gustavobflh worked well (now, in v3.4.0, the guilty lines in analyze.py are 1017 and 1026). So it seems that this is a fix that we want to have asap in the public version :-)

**Comment by dchooper** [comment]
Hi,

The bugfix is now in the pipeline and will be included in the next release, v3.5.0 (expected very soon).

Best,
Deanna
